### PR TITLE
[Feature](4/n) Support fast schema evolution in shared data mode

### PR DESCRIPTION
**Pull request by sduzh**
This PR aims to support global column unique id in shared data mode

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/lake/LakeTableHelper.java [hunk 1]
```diff
@@ -151,4 +153,19 @@ public static boolean isSharedPartitionDirectory(PhysicalPartition partition) th
     public static boolean isSharedDirectory(String path, long partitionId) {
         return !path.endsWith(String.format("/%d", partitionId));
     }
+
+    public static int restoreColumnUniqueId(OlapTable table) {
+        LOG.info("Restoring column unique ids of table '{}'(id={})", table.getName(), table.getId());
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/lake/LakeTableHelper.java:158]: Is this log necessary?

Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/lake/LakeTableHelper.java]: Removed

**Comment by sduzh** [comment]
@Mergifyio rebase

**Comment by sduzh** [comment]
@wyb @sevev @kevincai PTAL

**Comment by sduzh** [comment]
@Mergifyio rebase

**Review by sevev — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Pull request merged**